ARXIVNG-484

DavidLFielding opened the pull request:
This PR focuses on end of file cleanup and fixing carriage returns in files generated on PC and MAC platforms.

This cleanup is necessary because compilation fails when files are not cleaned up properly.

Jake indicates that unmacify running in production fails in certain cases. The Python implementation is new and improved from Perl implementation and may not experience similar failures. I will test this once I receive some test files from Jake.

JaimieMurdock commented on filemanager/process/upload.py:
```diff
@@ -999,10 +1015,140 @@ def _add_file(fpath: str, warnings: list, errors: list) -> File:
+        new_filepath = filepath + ".new"
+        with open(filepath, 'rb', 0) as infile, \
+            open(new_filepath, 'wb', 0) as outfile, \
+                mmap.mmap(infile.fileno(), 0, access=mmap.ACCESS_READ) as s:
```
Would you mind commenting a little on the decision to use `mmap.mmap` instead of `open` for the file `s` or why `s` is necessary given that its pointing to the same pointer as `infile`? Confused by the conventions, suspect I have something to learn having not used `mmap` before.

DavidLFielding replied:
The legacy code reads a file in blocks and I was looking for ways to eliminate some of the crufty code. I came about this solution for executing regexes over large files without reading file in chunks and this solution seems like an elegant way to do this. I will take another look at those references and try to follow up with a better explanation.

erickpeirson commented on filemanager/process/upload.py:
```diff
@@ -72,6 +75,19 @@ def __init__(self, upload_id: int):
+    def set_debug(self, set: bool):
+        """
+        Activate/deactivate debugging
+        :param set:
```
Don't forget to use numpy style docstrings, e.g. parameters are formatted differently

DavidLFielding replied:
I fell down on my documentation. I made a pass at improving docstrings in PR #28. I still have a few more files to cleanup. I will continue to improve on docstrings in file manager code.

erickpeirson commented on filemanager/process/upload.py:
```diff
@@ -999,10 +1015,140 @@ def _add_file(fpath: str, warnings: list, errors: list) -> File:
+        if self.debug():
+            self.log(f"Checking file termination for {filepath}.")
+
+        if os.path.exists(filepath):
```
One way to cut down on indentation here would be to return early. Ie 

```
if not os.path.exists(filepath):
    return
```

We'll want to aim for 79-character line limits as we get further along

DavidLFielding replied:
Fixed. I'll keep returning early in mind. I'll also work with pylint to eliminate other long lines.

erickpeirson commented on filemanager/process/upload.py:
```diff
@@ -999,10 +1015,140 @@ def _add_file(fpath: str, warnings: list, errors: list) -> File:
+        """
+
+        # Determine type of file we are dealing with PC or MAC
+        type = 'mac'
```
``type`` is a built-in; should probably use a different name

erickpeirson commented on filemanager/process/upload.py:
```diff
@@ -999,10 +1015,140 @@ def _add_file(fpath: str, warnings: list, errors: list) -> File:
+            open(new_filepath, 'wb', 0) as outfile, \
+                mmap.mmap(infile.fileno(), 0, access=mmap.ACCESS_READ) as s:
+
+            if type == 'pc':
```
It might be a good idea to define these values as globals in the module. Advantages are that it reduces the task of undetected typos, and is easier to see what the possible values are. E.g.

```python
PC = 'pc'
MAC = 'mac'

...

if file_type == PC:
    ...
```

DavidLFielding commented:
I've addressed the feedback above. There is a significant amount of doctoring enhancements in PR#28. Let me know if there are more changes need for this PR.

erickpeirson commented:
Thanks, looks great!
